File: CPP/BeaufortCipher.cpp
#include <iostream>
#include <vector>
using namespace std;

//O(n)
string beaufortCipher(string text, int k) {
    int n = text.length();
    string ciphertext = "";
    for (int i = 0; i < n; i++) {
        ciphertext += (90 - text[i] + k) % 26 + 65;
        k += (i + 1);
    }
    return ciphertext;
}

int main() {
    string text = "CAT";
    cout << text << endl;
    string encrypted_text = beaufortCipher(text, 1);
    cout << encrypted_text << endl;
    string text2 = "MOUSE";
    cout << text2 << endl;
    string encrypted_text2 = beaufortCipher(text2, 14);
    cout << encrypted_text2 << endl;
}

File: Python/BeaufortCipher.py
def beaufort_cipher(text, k):
    n = len(text)
    ciphertext = ""
    i = 0
    while i < n:
        ciphertext += chr((90 - ord(text[i]) + k) % 26 + 65)
        i += 1
        k += i
    return ciphertext

text = "CAT"
print(text)
print(beaufort_cipher(text, 1))
text2 = "MOUSE"
print(text2)
print(beaufort_cipher(text2, 14))